queryserver-orchestrator/src/main/java/org/apache/phoenix/queryserver/orchestrator/QueryServerCanaryOrchestrator.java

```diff
@@ -0,0 +1,144 @@
+                .help("Maximum time for " + "which the app should run before returning error. " +
+                        "default: 120 sec");
+
+        parser.addArgument("--testtable", "-tt").type(String.class).nargs("?").setDefault
+                ("PQSTEST").help("Custom " + "name for the test table. default: PQSTEST");
+
+        parser.addArgument("--logsinkclass", "-lsc").type(String.class).nargs("?").setDefault
+                ("org.apache.phoenix" + ".queryserver.tool.QueryServerCanaryTool$StdOutSink")
+                .help("Path to a Custom implementation for log " + "sink class. default: stdout");
+
+        parser.addArgument("--zkurl", "-zk").type(String.class).help("URL for Zookeeper");
+
+        parser.addArgument("--interval", "-in").type(String.class).nargs("?").setDefault("900")
+                .help("Time interval " + "between 2 consecutive test suite runs");
+
+        Namespace res = null;
+        try {
+            res = parser.parseArgs(args);
+        } catch (ArgumentParserException e) {
+            parser.handleError(e);
+        }
+        return res;
+    }
+
+    public static Map<String, String> getArgs(Namespace cArgs) {
+
+        Map<String, String> params = new HashMap<>();
+
+        for (Map.Entry<String, Object> entry : cArgs.getAttrs().entrySet()) {
+            params.put(entry.getKey(), (String) entry.getValue());
+        }
+        return params;
+    }
+
+    public static void main(String[] args) {
+
+        LOGGER.info("Starting PQS Canary Orchestrator...");
+        ToolWrapper tool = new ToolWrapper();
+        try {
+            Namespace cArgs = parseArgs(args);
+            if (cArgs == null) {
+                throw new RuntimeException("Argument parsing failed");
+            }
+
+            Map<String, String> params = getArgs(cArgs);
+            PATH = params.get("zkpath");
+
+            CuratorFramework curatorClient = CuratorFrameworkFactory.newClient(params.get("zkurl"), new
+                    BoundedExponentialBackoffRetry(1000, 60000, 50000));
+            curatorClient.start();
+
+            TestExecutorClient testExecutorClient = new TestExecutorClient(curatorClient, PATH, params, tool);
+            Thread testExecutorThread = new Thread (() -> {
+                testExecutorClient.start();
+            });
+            testExecutorThread.start();
+
+            Thread cleanupThread = new Thread(() -> {
+                LOGGER.info("Shutdown Hook for PQS Canary Orchestrator running...");
+                CloseableUtils.closeQuietly(testExecutorClient);
+                CloseableUtils.closeQuietly(curatorClient);
+                LOGGER.info("Closed Curator Client");
+            });
+            cleanupThread.setDaemon(true);
+            try {
+                testExecutorThread.join();
+            } catch (InterruptedException ex) {
+                LOGGER.error("The Main thread was interrupted", ex);
+            } finally {
+                cleanupThread.start();
```

Comment:
The main thread will exit here and will kill the cleanupThread too, since its a daemon thread. I think it's better to make this cleanup synchronous and spawning a new thread is not providing any benefit here.